### PR TITLE
Updated forecast operator schema for preprocessing

### PR DESCRIPTION
**Pull request by prasankh**
* Revised schema for preprocessing
```
preprocessing: 
 enabled: true
 steps:
   missing_value_imputation: true
   outlier_treatment: false
```
* updated test cases to cover multiple such scenarios

### CODE REVIEW

File: ads/opctl/operator/lowcode/forecast/schema.yaml [hunk 1]
```diff
@@ -286,11 +286,23 @@ spec:
       default: target
 
     preprocessing:
-      type: boolean
+      type: dict
       required: false
-      default: true
-      meta:
-        description: "preprocessing and feature engineering can be disabled using this flag, Defaults to true"
+      schema:
```
Comment (ahosler) [inline ads/opctl/operator/lowcode/forecast/schema.yaml:291]: Is the structure
preprocessing : {enabled, steps: {missing_val, outlier_d}}
or
 preprocessing : {enabled, missing_val, outlier_d}

Aren't the schema.yaml and operator_config disagreeing?

File: tests/operators/forecast/test_errors.py [hunk 2]
```diff
@@ -454,12 +456,14 @@ def split_df(df):
...
     yaml_i, output_data_path = populate_yaml(
         tmpdirname=tmpdirname,
         model=model,
         historical_data_path=historical_data_path,
         additional_data_path=additional_data_path,
         test_data_path=test_data_path,
+        preprocessing={"enabled": True, "steps": preprocessing_steps}
     )
     with pytest.raises(DataMismatchError):
         # 4 columns in historical data, but only 1 cat col specified
```
Comment (ahosler) [inline tests/operators/forecast/test_errors.py:469]: Can we add a test that reads the output file (forecast.csv) and asserts that the "input_value" column matches the historical data when "outlier_detection": False.
Then show that they are not equal when "outlier_detection": True

**Review by ahosler — CHANGES_REQUESTED** [review]
A couple of small follow up qs

File: ads/opctl/operator/lowcode/forecast/schema.yaml (hunk 1 above)
Comment (prasankh) [reply ads/opctl/operator/lowcode/forecast/schema.yaml:291]: The first one, updated schema.yaml

File: tests/operators/forecast/test_errors.py (hunk 2 above)
Comment (prasankh) [reply tests/operators/forecast/test_errors.py:514]: Done

**Review by ahosler — APPROVED** [review]

**Pull request closed without merging**